### DISCUSSION TITLE
C++ Port

### DISCUSSION THREAD

**Discussion by nlogozzo** [omansak/libvideo#245] [General]
Hi, I'm currently working on [Tube Converter](https://github.com/nlogozzo/NickvisionTubeConverter/tree/dev), a C++ application for downloading videos from YouTube. 

I would like to create a port of this library in C++ to use in the backend of my application. However, I'm new to creating a YouTube video downloader library of this sort and would like to understand how it works before I create a direct port without knowing what I'm looking at. 

Would @omansak (or anyone else) be able to share an outline of the algorithm/procedure used for downloading a video from YouTube? How does one get the link to the mp4 file to download? Get different resolution sizes? etc...

Thanks in advance.

**Comment by omansak** [comment]
Hi;

You can look this blog for algoritm :
https://tyrrrz.me/blog/reverse-engineering-youtube

**Reply by omansak** [reply]
its old but usefully

**Reply by nlogozzo** [reply]
Oh perfect thank you! I'll keep this open if I have any other questions as I go along.
